backend/app/modules/user/user.controller.ts

```diff
@@ -35,23 +35,53 @@ export class UserController {
 
   public async createUser(req: Request, res: Response): Promise<Response> {
     const body: IUser = req.body;
+    const { email, nickname } = body;
+    const isUserExistsByEmail = await this.UserService.getUserByEmail(email);
+    const isUserExistsByNickname = await this.UserService.getUserByNickname(nickname);
+
+    if(isUserExistsByEmail) {
+      throw new EmailExistsError('email');
+    }
+
+    if(isUserExistsByNickname) {
+      throw new UsernameExistsError('username')
+    }
+    
     const result = await this.UserService.createUser(body);
-    return new SuccessResponse<IUserResponse>(result).send(res)
+    return new SuccessResponse<IUserResponse>(result).send(res);
   }
 
   public async updateUser(req: Request, res: Response): Promise<Response> {
     const { id } = req.params;
     const body = req.body;
     const updatedUser = await this.UserService.updateUser(id, body);
-
+    console.log(req.params)
     if (!updatedUser || !updatedUser.matchedCount) {
       throw new NotFoundError('user');
     }
-
+    
     return new SuccessResponse<UpdateResult>(updatedUser).send(res)
   }
 
   public async deleteUser(req: Request, res: Response): Promise<Response> {
+    const { id } = req.params;
+    const body = req.body;
+    const updatedUser = await this.UserService.updateUser(
+        id, 
+        {
+          ...body, 
+          account_status: AccountStatus.DELETED,
+          account_status_change_date: Date.now(),
```

Comment:
why don't we use pre-hook for updating status_data_change

Reply:
the solution suggested by you is not working (im about set method in schema)( plak plak ... maybe it depends from mongoose version